api/internal/loctarget/util.go

```diff
@@ -80,6 +83,15 @@ func defaultNewDir(targetLdr ifc.Loader, spec *git.RepoSpec) string {
 	return strings.Join([]string{dstPrefix, targetDir}, "-")
 }
 
+// hasRef checks if path url has ref query string parameter
+func hasRef(path string) bool {
+	repoSpec, err := git.NewRepoSpecFromURL(path)
+	if err != nil {
+		log.Fatalf("%s: %s", "unable to parse validated root url", err.Error())
+	}
+	return repoSpec.Ref != ""
+}
+
```

Comment:
I'm down to move it, but can I clarify why it belongs with `RepoSpec`? 

This is a helper function for `localize` and doesn't add any functionality to `RepoSpec`. I didn't want to clutter `RepoSpec`'s method pool with a method that makes simple use of `RepoSpec`'s existing functionality. Also, the `log.Fatalf()` is specific to `localize`; we only call it because we call `hasRef` after loading path, during which a `RepoSpec` must have been created.

I originally wanted to put the `parseDomain` function (in this file) in `RepoSpec`, but it seemed to require refactoring the existing code, which (I believe?) wasn't recommended. https://github.com/kubernetes-sigs/kustomize/pull/4652#discussion_r937312000 I'm also down to refactor if time allows.